ikarus/python/finiteelements/material.hh

```diff
@@ -17,131 +17,188 @@
-      Dune::Python::insertClass<Ikarus::Materialname<args>>(                                                 \
-          m, #Materialname, Dune::Python::GenerateTypeName("Ikarus::" + std::string(#Materialname<##args>)), \
-          includes##Materialname)                                                                            \
-          .first;                                                                                            \
-  cls##Materialname.def(pybind11::init([](double emod, double nu) {                                          \
-                          auto matParameter =                                                                \
-                              Ikarus::toLamesFirstParameterAndShearModulus({.emodul = emod, .nu = nu});      \
-                          return new Materialname(matParameter);                                             \
-                        }),                                                                                  \
-                        "Material constructor that takes Young's modulus E and Poisson's ratio nu"           \
-                        "E"_a,                                                                               \
-                        "nu"_a);                                                                             \
-  MAKE_MaterialFunction(Materialname<##args>, storedEnergy);                                                 \
-  MAKE_MaterialFunction(Materialname<##args>, stresses);                                                     \
-  MAKE_MaterialFunction(Materialname<##args>, tangentModuli);
+  Ikarus::LamesFirstParameterAndShearModulus extractMaterialParameters(const pybind11::kwargs& kwargs) {
+    // clang-format off
+    static const std::map<std::pair<std::string, std::string>, std::function<LamesFirstParameterAndShearModulus(const pybind11::kwargs&)>> conversionMap = {
+      {{"E", "nu"}, [](const auto& kw){ return convertMaterialParameters<YoungsModulusAndPoissonsRatio>(kw, "E", "nu"); }},
+      {{"E", "mu"}, [](const auto& kw){ return convertMaterialParameters<YoungsModulusAndShearModulus>(kw, "E", "mu"); }},
+      {{"E", "K"}, [](const auto& kw){ return convertMaterialParameters<YoungsModulusAndBulkModulus>(kw, "E", "K"); }},
+      {{"E", "Lambda"}, [](const auto& kw){ return convertMaterialParameters<YoungsModulusAndLamesFirstParameter>(kw, "E", "Lambda"); }},
+      {{"K", "Lambda"}, [](const auto& kw){ return convertMaterialParameters<BulkModulusAndLamesFirstParameter>(kw, "K", "Lambda"); }},
+      {{"Lambda", "mu"}, [](const auto& kw){ return LamesFirstParameterAndShearModulus{kw["Lambda"].template cast<double>(), kw["mu"].template cast<double>()}; }}
+    };
+    // clang-format on
 
-namespace Ikarus::Python {
+    if (kwargs.size() != 2)
+      DUNE_THROW(Dune::IOError, "The number of material parameters passed to the material should be 2");
+
+    for (const auto& conversion : conversionMap) {
```

Comment:
Somehow my comment here is gone. Did you check what I suggested here or didn't it go here at all

Reply:
Yeah i couldn't get this to work
see my comment here 
https://github.com/ikarus-project/ikarus/pull/317#discussion_r1770876595